hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOmSnapshot.java

```diff
@@ -916,15 +918,31 @@ public void testUniqueSnapshotId()
   public void testSnapshotOpensWithDisabledAutoCompaction() throws Exception {
     String snapPrefix = createSnapshot(volumeName, bucketName);
     RDBStore snapshotDBStore = (RDBStore)
-            ((OmSnapshot)cluster.getOzoneManager().getOmSnapshotManager()
+        ((OmSnapshot)cluster.getOzoneManager().getOmSnapshotManager()
             .checkForSnapshot(volumeName, bucketName, snapPrefix))
             .getMetadataManager().getStore();
 
     for (String table : snapshotDBStore.getTableNames().values()) {
       Assertions.assertTrue(snapshotDBStore.getDb().getColumnFamily(table)
-              .getHandle().getDescriptor()
-              .getOptions().disableAutoCompactions());
+          .getHandle().getDescriptor()
+          .getOptions().disableAutoCompactions());
     }
   }
 
+  @Test
+  public void testCompactionDagDisableForSnapshotMetadata() throws Exception {
+    String snapshotName = createSnapshot(volumeName, bucketName);
+
+    // RocksDBCheckpointDiffer should be not null for active store metadata.
+    assertNotNull(cluster.getOzoneManager().getMetadataManager().getStore()
+        .getRocksDBCheckpointDiffer());
+
+    OmSnapshot omSnapshot = (OmSnapshot) cluster.getOzoneManager()
+        .getOmSnapshotManager()
+        .checkForSnapshot(volumeName, bucketName, snapshotName);
+
+    // RocksDBCheckpointDiffer should be null for snapshot metadata.
+    assertNull(omSnapshot.getMetadataManager().getStore()
```

Comment:
better thing to assert would be to check if there are no listeners attached to the dboptions.

Assertions.assertEquals(omSnapshot.getMetadataManager().getStore().getListeners().size(), 0);

Reply:
This will also check if there are any other listeners attached.

Reply:
Updated source and test code to assert listener attached to `DBOptions`.